models.py

```diff
@@ -63,7 +62,9 @@ def forward(self, inputs, hidden):
         """
         hidden = self.h2h(hidden) + self.i2h(inputs)
         hidden = self.non_linearity(hidden)
-
+        # no gradient is computed if we don't call requires_grad_(True)
```

Comment:
for 5.2 we need to compute the gradient with respect to the hidden state. So i activate the gradient here